<launch>
  <node name="object_singulation_perception" pkg="tabletop_pushing" type="object_singulation_node" respawn="false" output="screen">
    <remap from="color_image_topic" to="/kinect_head/rgb/image_color"/>
    <remap from="depth_image_topic" to="/kinect_head/depth/image"/>
    <remap from="point_cloud_topic" to="/kinect_head/depth/points"/>

    <param name="display_wait_ms" value="3" type="int"/>
    <param name="segmenting_moving_stuff" value="false" type="bool"/>
    <param name="autostart_tracking" value="true" type="bool"/>
    <param name="autostart_pcl_segmentation" value="true" type="bool"/>

    <param name="crop_min_x" value="22" type="int"/>
    <param name="crop_max_x" value="600" type="int"/>
    <param name="crop_min_y" value="32" type="int"/>
    <param name="crop_max_y" value="469" type="int"/>
    <param name="image_hist_size" value="2" type="int"/>

    <param name="workspace_frame" value="/torso_lift_link" type="string"/>
    <param name="min_workspace_x" value="-1.0" type="double"/>
    <param name="max_workspace_x" value="1.75" type="double"/>
    <param name="min_workspace_y" value="-1.2" type="double"/>
    <param name="max_workspace_y" value="1.2" type="double"/>
    <param name="min_workspace_z" value="-0.8" type="double"/>
    <param name="max_workspace_z" value="0.6" type="double"/>
    <param name="below_table_z" value="0.1" type="double"/>

    <param name="mgc_workspace_background_weight" value="0.95" type="double"/>
    <param name="mgc_min_weight" value="0.01" type="double"/>
    <param name="mgc_flow_gain" value="0.35" type="double"/>
    <param name="mgc_table_var" value="0.03" type="double"/>
    <param name="mgc_arm_dist_var" value="15.0" type="double"/>

    <param name="mgc_w_c_alpha" value="0.4" type="double"/>
    <param name="mgc_w_c_beta" value="0.9" type="double"/>
    <param name="mgc_w_c_gamma" value="2.0" type="double"/>

    <param name="mgc_magnitude_thresh" value="1.0" type="double"/>
    <param name="mgc_arm_grow_radius" value="1" type="int"/>
    <param name="mgc_arm_search_radius" value="100" type="int"/>
    <param name="mgc_arm_color_var_add" value="0.1" type="double"/>
    <param name="mgc_arm_color_weight" value="0.8" type="double"/>
    <param name="mgc_arm_dist_weight" value="0.5" type="double"/>

    <param name="lk_win_size" value="5" type="int"/>
    <param name="lk_num_levels" value="4" type="int"/>
    <param name="lk_ratio_thresh" value="15.0" type="double"/>
    <param name="num_downsamples" value="1" type="int"/>

    <param name="arm_detection_theta" value="10.0" type="double"/>
    <param name="max_flow_clusters" value="7" type="int"/>
    <param name="flow_cluster_max_iter" value="2000" type="int"/>
    <param name="flow_cluster_epsilon" value="0.1" type="double"/>
    <param name="flow_cluster_attempts" value="5" type="int"/>
    <param name="affine_RANSAC_epsilon" value="2.0" type="double"/>
    <param name="surf_hessian_thresh" value="50.0" type="double"/>
    <param name="use_fast_corners" value="true" type="bool"/>
    <param name="min_affine_point_set_size" value="200" type="int"/>
    <param name="max_ransac_iters" value="200" type="int"/>
    <param name="minimum_new_cluster_separation" value="5.0" type="double"/>

    <param name="pcl_cluster_tolerance" value="0.02" type="double"/>
    <param name="pcl_min_cluster_size" value="50" type="int"/>
    <param name="pcl_max_cluster_size" value="2500" type="int"/>
    <param name="pcl_voxel_downsample_res" value="0.01" type="double"/>
    <param name="use_pcl_voxel_downsample" value="false" type="bool"/>
    <param name="table_ransac_thresh" value="0.015" type="double"/>
    <param name="pcl_difference_thresh" value="0.01" type="double"/>

    <param name="min_table_z" value="-1.0" type="double"/>
    <param name="max_table_z" value="1.0" type="double"/>
    <param name="min_pushing_y" value="-1.0" type="double"/>
    <param name="max_pushing_y" value="1.0" type="double"/>
    <param name="min_pushing_x" value="0.0" type="double"/>
    <param name="max_pushing_x" value="2.0" type="double"/>

  </node>
</launch>
